blog/2023-11-15-koordinates-and-stately-case-study/index.mdx

```diff
@@ -0,0 +1,67 @@
+<!--truncate-->
+Unlike competing products that require vendor lock-in and lack compatibility across the entire GIS stack, Koordinates’ offerings are modular: users may use as much or as little of the tooling as they like, and the tooling is interoperable with existing GIS environments.
+
+![Koordinates Logo](./koordinates_logo_rev.png)
+
+## UI state management
+
+Koordinates provide their users with both a web app, named Koordinates Cloud, as well as an Electron-based desktop app. In order to filter through and query such large datasets, the team naturally needed to expose a lot of controls for end-user functionality. The frontends for the apps allow end users to accomplish complex tasks, like:
+
+- **searching** for the **right data via the Browser
+- **visualizing** relevant geospatial information in the Map Viewer
+- **automating** data management with workflows in the Data Manager
+
+Managing context and application state was a core concern for the team, and for determining the best framework to manage complex state, they turned to XState.
+
+## Repository State Management
+
+The desktop app Koordinates provides also ships with **[Kart](https://github.com/koordinates/kart),** an open source version control platform built specifically for large geospatial datasets. Being built on top of Git, Kart provides a familiar user experience for developers used to the version control system. Specifically, Kart allows users to:
+
+- **clone** geospatial data to users’ local machines
+- **manage** branches for their datasets - creating, merging, and deleting them
+- **roll back** branches to previous commits, preserving changes made over time
+
+In the case of Kart, many of these operations can be long-running and have **several** edge cases, making Stately’s tooling to visualize, understand, and execute logic from stateful diagrams a massive benefit.
+
+## Breaking complex UI’s into manageable components **
+
+In the case of both their products, Koordinates was able to use Stately’s tooling to break their “big” problems down into several, distinct smaller ones. Taylor Lodge, an engineer on the core team, explained that with Stately’s visual editor and simulation tooling, they were able to break their application logic into several small, testable state machines. For example, the UI for Koordinates Cloud has several components, but the logic for loading elements in their list view is represented rather succinctly in this state machine: 
+
+
+<EmbedMachine
+  name="Koordinates List View Machine"
+  embedURL="https://stately.ai/registry/editor/1b48bbf1-6f7a-4847-b1e0-47ee5c283371?mode=design&machineId=68b835cc-49fc-4b9c-972e-a63c01465c1f"/>
```

Comment:
```suggestion
  embedURL="https://stately.ai/registry/editor/embed/1b48bbf1-6f7a-4847-b1e0-47ee5c283371?machineId=68b835cc-49fc-4b9c-972e-a63c01465c1f&mode=design"/>
```

Reply:
I suggest changing this and the next link to be an embed URL.

Reply:
Ah, I thank you for this, @mellson. Didn't know about these parameters.